From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 8 Dec 2020 13:37:07 +0100
Subject: backends/native: Drop MetaEventNative

This doesn't really hold anything now, so we can stop trying to
copy these across events.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1623>
---
 src/backends/native/meta-backend-native.c |  1 -
 src/backends/native/meta-event-native.c   | 54 -------------------------------
 src/backends/native/meta-event-native.h   | 45 --------------------------
 src/backends/native/meta-seat-impl.c      |  1 -
 src/backends/native/meta-seat-native.c    | 11 -------
 src/meson.build                           |  2 --
 6 files changed, 114 deletions(-)
 delete mode 100644 src/backends/native/meta-event-native.c
 delete mode 100644 src/backends/native/meta-event-native.h

diff --git a/src/backends/native/meta-backend-native.c b/src/backends/native/meta-backend-native.c
index 7206cb6..f34b1d2 100644
--- a/src/backends/native/meta-backend-native.c
+++ b/src/backends/native/meta-backend-native.c
@@ -50,7 +50,6 @@
 #include "backends/meta-settings-private.h"
 #include "backends/meta-stage-private.h"
 #include "backends/native/meta-clutter-backend-native.h"
-#include "backends/native/meta-event-native.h"
 #include "backends/native/meta-kms.h"
 #include "backends/native/meta-kms-device.h"
 #include "backends/native/meta-launcher.h"
diff --git a/src/backends/native/meta-event-native.c b/src/backends/native/meta-event-native.c
deleted file mode 100644
index c273176..0000000
--- a/src/backends/native/meta-event-native.c
+++ /dev/null
@@ -1,54 +0,0 @@
-/*
- * Copyright (C) 2015 Red Hat
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 2 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library. If not, see <http://www.gnu.org/licenses/>.
- *
- * Authored by:
- *      Carlos Garnacho <carlosg@gnome.org>
- */
-
-#include "config.h"
-
-#include "backends/native/meta-event-native.h"
-#include "backends/native/meta-input-thread.h"
-#include "clutter/clutter-mutter.h"
-
-typedef struct _MetaEventNative MetaEventNative;
-
-struct _MetaEventNative
-{
-  uint64_t time_usec;
-
-  gboolean has_relative_motion;
-  double dx;
-  double dy;
-  double dx_unaccel;
-  double dy_unaccel;
-};
-
-MetaEventNative *
-meta_event_native_copy (MetaEventNative *event_evdev)
-{
-  if (event_evdev != NULL)
-    return g_slice_dup (MetaEventNative, event_evdev);
-
-  return NULL;
-}
-
-void
-meta_event_native_free (MetaEventNative *event_evdev)
-{
-  if (event_evdev != NULL)
-    g_slice_free (MetaEventNative, event_evdev);
-}
diff --git a/src/backends/native/meta-event-native.h b/src/backends/native/meta-event-native.h
deleted file mode 100644
index 43cf7b9..0000000
--- a/src/backends/native/meta-event-native.h
+++ /dev/null
@@ -1,45 +0,0 @@
-/*
- * Copyright (C) 2015 Red Hat
- *
- * This library is free software; you can redistribute it and/or
- * modify it under the terms of the GNU Lesser General Public
- * License as published by the Free Software Foundation; either
- * version 2 of the License, or (at your option) any later version.
- *
- * This library is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public
- * License along with this library. If not, see <http://www.gnu.org/licenses/>.
- *
- * Authored by:
- *      Carlos Garnacho <carlosg@gnome.org>
- */
-
-#ifndef META_EVENT_NATIVE_H
-#define META_EVENT_NATIVE_H
-
-#include "clutter/clutter.h"
-
-typedef struct _MetaEventNative MetaEventNative;
-
-MetaEventNative * meta_event_native_copy (MetaEventNative *event_evdev);
-void              meta_event_native_free (MetaEventNative *event_evdev);
-
-uint64_t          meta_event_native_get_time_usec  (const ClutterEvent *event);
-void              meta_event_native_set_time_usec  (ClutterEvent *event,
-                                                    uint64_t      time_usec);
-void              meta_event_native_set_relative_motion (ClutterEvent *event,
-                                                         double        dx,
-                                                         double        dy,
-                                                         double        dx_unaccel,
-                                                         double        dy_unaccel);
-gboolean          meta_event_native_get_relative_motion (const ClutterEvent *event,
-                                                         double             *dx,
-                                                         double             *dy,
-                                                         double             *dx_unaccel,
-                                                         double             *dy_unaccel);
-
-#endif /* META_EVENT_NATIVE_H */
diff --git a/src/backends/native/meta-seat-impl.c b/src/backends/native/meta-seat-impl.c
index 21d5098..09bef5e 100644
--- a/src/backends/native/meta-seat-impl.c
+++ b/src/backends/native/meta-seat-impl.c
@@ -35,7 +35,6 @@
 
 #include "backends/meta-cursor-tracker-private.h"
 #include "backends/native/meta-barrier-native.h"
-#include "backends/native/meta-event-native.h"
 #include "backends/native/meta-input-thread.h"
 #include "backends/native/meta-virtual-input-device-native.h"
 #include "clutter/clutter-mutter.h"
diff --git a/src/backends/native/meta-seat-native.c b/src/backends/native/meta-seat-native.c
index f56be66..d691c52 100644
--- a/src/backends/native/meta-seat-native.c
+++ b/src/backends/native/meta-seat-native.c
@@ -31,7 +31,6 @@
 #include "backends/meta-cursor-tracker-private.h"
 #include "backends/meta-keymap-utils.h"
 #include "backends/native/meta-barrier-native.h"
-#include "backends/native/meta-event-native.h"
 #include "backends/native/meta-input-thread.h"
 #include "backends/native/meta-keymap-native.h"
 #include "backends/native/meta-virtual-input-device-native.h"
@@ -294,22 +293,12 @@ meta_seat_native_copy_event_data (ClutterSeat        *seat,
                                   const ClutterEvent *src,
                                   ClutterEvent       *dest)
 {
-  MetaEventNative *event_evdev;
-
-  event_evdev = _clutter_event_get_platform_data (src);
-  if (event_evdev != NULL)
-    _clutter_event_set_platform_data (dest, meta_event_native_copy (event_evdev));
 }
 
 static void
 meta_seat_native_free_event_data (ClutterSeat  *seat,
                                   ClutterEvent *event)
 {
-  MetaEventNative *event_evdev;
-
-  event_evdev = _clutter_event_get_platform_data (event);
-  if (event_evdev != NULL)
-    meta_event_native_free (event_evdev);
 }
 
 static guint
diff --git a/src/meson.build b/src/meson.build
index 77f2748..b597e29 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -652,8 +652,6 @@ if have_native_backend
     'backends/native/meta-drm-buffer-import.h',
     'backends/native/meta-drm-buffer.c',
     'backends/native/meta-drm-buffer.h',
-    'backends/native/meta-event-native.c',
-    'backends/native/meta-event-native.h',
     'backends/native/meta-gpu-kms.c',
     'backends/native/meta-gpu-kms.h',
     'backends/native/meta-input-device-native.c',
